### PR TITLE
ozi: don't deinit a write stream if we don't have one.

### PR DESCRIPTION
**Pull request by robertlipe**
We can get into close() with a null stream_ TextStream. I think it has to do with ozi_openfile() reopening a file with a new name to match an extension of a certain type so it's renaming a file that hasn't been written. See issue #646

### CODE REVIEW

File: ozi.cc [hunk 1]
```diff
@@ -172,6 +172,9 @@ ozi_open_io(const QString& fname, QIODevice::OpenModeFlag mode)
 static void
 ozi_close_io()
 {
+  if (!stream) {
```
Comment (tsteven4) [inline ozi.cc:175]: This makes the check around the call to ozi_close_io in ozi_openfile redundant (line 263).

**Comment by tsteven4** [comment]
We should move this along, with or without resolving the redundancy I pointed out above.  It's a real bug and a good fix.

**Comment by tsteven4** [comment]
> 
> 
> We can get into close() with a null stream_ TextStream. I think it has to do with ozi_openfile() reopening a file with a new name to match an extension of a certain type so it's renaming a file that hasn't been written. See issue #646

The only case I have found that generates the fault is when waypt_count, route_count and track_count are all zero.  I don't think there is an issue with reopening a file with a new name as speculated above.

I have enhancements to the ozi test that covers both of the above cases.  I can push those to this branch or merge this branch and put them in a separate PR.   @robertlipe please tell me how you would like to proceed.  thanks

**Comment by tsteven4** [comment]
In the absence of further communication I am merging this.  The test cases demonstrate resolution of the reported segmentation fault.

**Comment by robertlipe** [comment]
Agreed. Thank you. That sat almost at the top of my TODO list for too long.

Sidebar for anyone else with too many computers and too many source trees in your life: include the computer and the directory name when creating git branches.

**Pull request merged**